org.eclipse.scanning.test/src/org/eclipse/scanning/test/epics/EpicsV4ConnectorTest.java

```diff
@@ -46,27 +47,33 @@
  */
 public class EpicsV4ConnectorTest {
 
-	private IMalcolmService      service;
+	private IScanService service;
 	private IEPICSv4Device epicsv4Device;
+	private EpicsV4ConnectorService connectorService;
 
 	@Before
 	public void before() throws Exception {
 		// The real service, get it from OSGi outside this test!
 		// Not required in OSGi mode (do not add this to your real code GET THE SERVICE FROM OSGi!)
-		this.service = new MalcolmService(new EpicsV4ConnectorService(), null);
+		this.connectorService = new EpicsV4ConnectorService();
+		this.service = new RunnableDeviceServiceImpl();
 	}
 
 	@After
 	public void after() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)